File: library-app/src/models/AddBookRequest.ts
class AddBookRequest {
    title: string;
    author: string;
    description: string;
    copies: number;
    category: string;
    image?: string;

    constructor(title: string, author: string, description: string, copies: number, category: string, image: string) {
        this.title = title;
        this.author = author;
        this.description = description;
        this.copies = copies;
        this.category = category;
        this.image = image;
    }
}

export default AddBookRequest;

File: library-server/src/main/java/com/ivaalsolutions/libraryserver/requestmodels/AddBookRequest.java
package com.ivaalsolutions.libraryserver.requestmodels;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class AddBookRequest {
    private String title;
    private String author;
    private String description;
    private int copies;
    private String category;
    private String image;
}
